### PR TITLE
Disable checkboxes programatically through custom method

### PR DESCRIPTION
**Pull request by jvmonjo**
Changed the PR #510 with this new one, which makes no assumption about client data and uses custom method

### CODE REVIEW

**Comment by jtommy** [comment]
@jvmonjo Thanks! PR is great but to be honest i don't like the prop name. What do you think of `isRowCheckable` ? Did you have other proposals ?

**Comment by jvmonjo** [comment]
I think `isRowCheckable ` makes total sense. I'll change it right now.

**Comment by jtommy** [comment]
@jvmonjo Please resolve conflict and update api docs

**Comment by jvmonjo** [comment]
Actually. Giving it a second thought. Shoudn't `isRowCheckable` return `false` if the checkbox is disabled?

I think it must be the other way around. It has to be a question which when you respond `true`it means that is disabled.

**Comment by jvmonjo** [comment]
So maybe `isRowDisabled` ?

**Comment by jtommy** [comment]
You might change default value. 
I don't like `isRowDisabled` because it's generic about row and not about checked mode.

**Comment by jvmonjo** [comment]
Don't merge yet. I need to double check the new default.

**Comment by jvmonjo** [comment]
I've just fixed the logic to the new prop default. I think it's now safe to merge if you want.

File: src/components/table/Table.vue [hunk 1]
```diff
@@ -160,6 +160,7 @@
 </template>
 
 <script>
+/* eslint-disable */
```
Comment (jtommy) [inline src/components/table/Table.vue]: Why that ?

File: src/components/table/Table.vue [hunk 2]
```diff
@@ -295,7 +296,7 @@
              */
             isAllChecked() {
                 const validVisibleData = this.visibleData.filter(
-                        (row) => this.isRowCheckable(row) !== true)
+                        (row) => this.isRowCheckable(row) == true)
```
Comment (jtommy) [inline src/components/table/Table.vue]: Is not enough only `this.isRowCheckable(row)` ?

**Comment by jvmonjo** [comment]
Sorry about eslint flag. I had to do it to run de dev server. I've already remove it

File: src/components/table/Table.vue (hunk 1 above)
Comment (jvmonjo) [inline src/components/table/Table.vue]: Sorry. I had to do it to be able to run my dev server without eslint complaining. I've already remove it.

File: src/components/table/Table.vue (hunk 2 above)
Comment (jvmonjo) [reply src/components/table/Table.vue]: How would you do it? Feel free to rewrite it in a simpler way

**Comment by jtommy** [comment]
Please check review and fix eslint rule error

File: src/components/table/Table.vue (hunk 2 above)
Comment (jvmonjo) [reply src/components/table/Table.vue]: I get it. I already changed it

**Pull request merged**